osu.Game/Rulesets/Difficulty/Skills/Skill.cs

```diff
@@ -28,12 +36,32 @@ public abstract class Skill
         protected Skill(Mod[] mods)
         {
             this.mods = mods;
+
+            if (PreviousCollectionSoftCapacity > 0)
```

Comment:
Nothing wrong with having a capacity of 1 right? Might as well keep it like that by default and have this always instantiated.

Reply:
Yea this was just to be super conscious of unnecessary allocations.
Since this only happens once per skill, it's probably not worth the optimisation.
The big part is just leaving the `RemoveExtraneousHistory()` and `AddToHistory()` methods empty.

Reply:
Ah right, it'd be empty... Do you have anything against moving this to `StrainSkill` and keeping the implementation private, making with the previous implementation (keep a soft history 1 + private add/remove history methods).

Your PP+ skills could use a local `ReverseQueue` with their own special handling.